source/modulo_components/src/ComponentInterface.cpp

```diff
@@ -37,7 +37,17 @@ ComponentInterface::ComponentInterface(
 
   this->step_timer_ = rclcpp::create_wall_timer(
       std::chrono::nanoseconds(static_cast<int64_t>(this->get_parameter_value<double>("period") * 1e9)),
-      [this] { this->step(); }, nullptr, this->node_base_.get(), this->node_timers_.get());
+      [this] {
+        if (this->step_mutex_.try_lock()) {
+          this->step();
```

Comment:
comment(non-blocking): If step had an exception, I think the wall timer callback might fail but try again on the next timer, in which case the mutex would stay locked (or maybe it kills the whole executor, in which case the mutex is not a concern anymore). I am not sure what rclcpp does with unhandled exceptions in wall timer callbacks and it would be nice if this could be tested.

But in any case, this should be safe (mutex always unlocks) because we `try catch` exceptions inside step. It's just a bit hidden from the outside.

Reply:
Yeah I asked myself the same question when I implemented it, I'll see if it can be tested easily

Reply:
So I've tested (both cpp and python) components that dont handle exceptions on step and the result in both cases is that ROS just dies. So as long as we're catching all exceptions in the wall timers I think we are good and can't run into the issue mentioned above.